libsolidity/analysis/ViewPureChecker.cpp

```diff
@@ -274,8 +276,8 @@ void ViewPureChecker::reportMutability(
 				m_errorReporter.typeError(
 					5887_error,
 					_location,
-					"\"msg.value\" and \"callvalue()\" can only be used in payable public functions. Make the function "
-					"\"payable\" or use an internal function to avoid this error."
+					"\"msg.value\" and \"callvalue()\" can only be used in " + mutability + " " + kind + "s. Make the " + kind + " " +
+					"\"payable\"" +  (m_currentFunction->isConstructor() ? "." : " or use an internal " + kind + " to avoid this error.")
```

Comment:
I think it's just easier to have two error messages.

Reply:
thats true :)